core/src/autogluon/core/models/ensemble/stacker_ensemble_model.py

```diff
@@ -229,3 +229,10 @@ def _add_stack_to_feature_metadata(self):
                 stacker_feature_metadata = stacker_feature_metadata.keep_features([feature for feature in existing_features if feature in type_map_raw])
                 if set(stacker_feature_metadata.get_features()) != set(existing_stack_features):
                     self.feature_metadata = self.feature_metadata.add_special_types(stacker_feature_metadata.get_type_map_special())
+
+    def compile(self, compiler_configs=None) -> str:
+        if compiler_configs is None:
+            compiler_configs = {}
+        for model_name in self.base_model_names:
+            model = self.load_base_model(model_name)
+            model.compile(compiler_configs=compiler_configs)
```

Comment:
Not correct. Must be fixed before merging. Do not edit stacker_ensemble_model. Edit BaggedEnsembleModel.

for `model_name in self.models`. Follow predict_proba logic for bagged ensemble model.

Reply:
I still cannot follow predict_proba logic for bagged ensemble model, because:
1. BaggedEnsemableModel doesn’t have the attribute base_model_paths_dict, therefore, won’t be able to load the base models for compilation.
2. The predict_proba method in BaggedEnsembleModel only takes outputs of the base models. The outputs of the base models are calculated in StackerEnsembleModel.preprocess, where it takes base_model_paths_dict attribute to load the base model.

Here are some details:
```python
def compile(self, compiler_configs=None):
    if compiler_configs is None:
        compiler_configs = {}
    # Following predict_proba logic for bagged ensemble model
    for model in self.models:
        # model is a GreedyWeightedEnsembleModel
        model = self.load_child(model)
        # model is still a GreedyWeightedEnsembleModel
        # NOTE: GreedyWeightedEnsembleModel would only weight outputs of base models
       
        # We need to compile the base model here, in order to compile it.
        model.compile(compiler_configs=compiler_configs)
```
```python
def predict_proba(self, X, normalize=None, **kwargs):
    model = self.load_child(self.models[0])
    # The input X is a pd.DataFrame
    # For a WeightedEnsembleModel, this will call into preprocess method in StackerEnsembleModel.
    # Such preprocess method takes base_model_paths_dict to load base model
    X = self.preprocess(X, model=model, **kwargs)
    # The output X is a list of np array
   
    pred_proba = model.predict_proba(X=X, preprocess_nonadaptive=False, normalize=normalize)
    for model in self.models[1:]:
        model = self.load_child(model)
        pred_proba += model.predict_proba(X=X, preprocess_nonadaptive=False, normalize=normalize)
    pred_proba = pred_proba / len(self.models)
```
In summary, We need to load the base model in BaggedEnsembleModel, in order to compile it. However, it is the StackerEnsembleModel that keeps the base_model_paths_dict attribute, which is required to load the base model.

Reply:
As per offline discussion, we can have access to base models by introducing `with_ancestors` argument in `compile_models` interface. Such that the base models are loaded by abstract_trainer and get compiled there.

Reply:
Added `with_ancestors` argument in abstract_trainer, in order to compile base models.